framework-cloud/cloud-deployment-plugin/src/main/java/org/kie/cloud/plugin/Undeploy.java

```diff
@@ -38,18 +40,19 @@
     private MavenProject mavenProject;
 
     @Override public void execute() throws MojoExecutionException, MojoFailureException {
-        DeploymentScenarioBuilderFactory deploymentScenarioBuilderFactory = DeploymentScenarioBuilderFactory.getInstance();
-        deploymentScenarioBuilderFactory.deleteNamespace(getNamespace());
+        DeploymentScenarioBuilderFactory deploymentScenarioBuilderFactory = DeploymentScenarioBuilderFactoryLoader
+                .getInstance(loadCloudProperties().getProperty(CLOUD_API_IMPLEMENTATION_PROPERTY));
+        deploymentScenarioBuilderFactory.deleteNamespace(loadCloudProperties().getProperty(NAMESPACE_PROPERTY));
     }
 
-    private String getNamespace() {
+    private Properties loadCloudProperties() {
```

Comment:
Separate class for plugin settings was created - "CloudDeploymentPluginConfiguration".